examples/window-app/silabs/efr32/include/WindowManager.h

```diff
@@ -0,0 +1,251 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/clusters/window-covering-server/window-covering-server.h>
+#include <lib/core/CHIPError.h>
+
+#include <FreeRTOS.h>
+#include "LEDWidget.h"
+
+#include <queue.h>
+#include <setup_payload/QRCodeSetupPayloadGenerator.h>
+#include <string>
+#include <task.h>
+#include <timers.h>
+#ifdef DISPLAY_ENABLED
+#include <LcdPainter.h>
+#endif
+
+using namespace chip::app::Clusters::WindowCovering;
+
+class WindowManager
+{
+public:
+
+    static WindowManager sInstance;
+    struct Timer
+    {
+        typedef void (*Callback)(Timer & timer);
+
+        Timer(const char * name, uint32_t timeoutInMs, Callback callback, void * context);
+
+        void Start();
+        void Stop();
+        void Timeout();
+
+        const char * mName    = nullptr;
+        uint32_t mTimeoutInMs = 0;
+        Callback mCallback    = nullptr;
+        void * mContext       = nullptr;
+        bool mIsActive        = false;
+
+        TimerHandle_t mHandler = nullptr;
+        void IsrStart();
+
+    private:
+        static void TimerCallback(TimerHandle_t xTimer);
+    };
+
+    struct Button
```

Comment:
Is this struct really needed?

Reply:
The struct is used within the WindowManager to define the actions of the buttons on the board. Maybe the buttons behavior could be called in AppTask, like the lightning app example ?